### ISSUE TITLE
Nested Select in mongodb

### ISSUE THREAD

**Issue by judemanutd** [typeorm/typeorm#3266]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I am trying to do a nested select for fetching out particular field of a nested array of objects in mongodb. Currently the select that worked for me looks like this.

```ts 
    const data = await StudentRepository.find({
      select: ["id", "name"],
    });
```

The structure of the object in mongo looks like this

```json
{  
   "_id":"5c07b4bac10fcb38083d39b5",
   "name":"Tim",
   "address":[  
      {  
         "pincode":123,
         "other":"lorem ipsum"
      },
      {  
         "pincode":456,
         "other":"lorem ipsum"
      }
   ]
}
```

Here `address` is an array contained inside the `student object`

I would like to do something like this 

```ts 
    const data = await StudentRepository.find({
      select: ["id", "name","address.pincode"],
    });
```

The final structure that I am trying to achieve is 

```json
{  
   "_id":"5c07b4bac10fcb38083d39b5",
   "name":"Tim",
   "address":[  
      {  
         "pincode":123
      },
      {  
         "pincode":456
      }
   ]
}
```

**Comment by rustamwin** [comment]
> const data = await StudentRepository.find({
      select: ["id", "name","address.pincode"],
    });

this way doesn't support for collection queries, use `aggregation` instead

**Comment by rustamwin** [comment]
Duplicate of #2789

**Comment by vlapo** [comment]
I do not think so this is duplicate of https://github.com/typeorm/typeorm/issues/2789. This issue is about selecting/projection and https://github.com/typeorm/typeorm/issues/2789 is about query by nested fields.

**Comment by rustamwin** [comment]
Ahh sorry, see #3756
